dlib/test/dnn.cpp

```diff
@@ -4453,47 +4486,6 @@ namespace
         }
     }
 
-// ----------------------------------------------------------------------------------------
-
-    void test_tril()
-    {
-        print_spinner();        
-        using net_type = tag1<tril_mask<tag2<input<matrix<float>>>>>;
-        net_type net;
-
-        // Input tensor
-        dlib::rand rnd;
-        const int nr = 2, nc = 3;
-        constexpr int n_samples = 3, k = 1;
-        std::vector<matrix<float>> x(n_samples);
-        matrix<float> xtmp(nr, nc);
-        for (int ii = 0; ii < n_samples; ++ii) {
-            for (int jj = 0; jj < nr; ++jj)
-                for (int kk = 0; kk < nc; ++kk)
-                    xtmp(jj, kk) = rnd.get_random_gaussian();
-            x[ii] = xtmp;
-        }
-
-        // Convert input matrix to tensor
-        resizable_tensor input_tensor;
-        net.to_tensor(&x[0], &x[0] + n_samples, input_tensor);
-        net.forward(input_tensor);
-
-        // Expected output tensor (manually set for comparison)
-        resizable_tensor expected_output;
-        expected_output.copy_size(input_tensor);
-        tt::copy_tensor(false, expected_output, 0, input_tensor, 0, input_tensor.k());
-        for (int ii = 0; ii < n_samples; ++ii) {
-            expected_output.host()[tensor_index(expected_output, ii, 0, 0, 1)] = -std::numeric_limits<float>::infinity();
-            expected_output.host()[tensor_index(expected_output, ii, 0, 0, 2)] = -std::numeric_limits<float>::infinity();
-            expected_output.host()[tensor_index(expected_output, ii, 0, 1, 2)] = -std::numeric_limits<float>::infinity();
-        }
-
-        // Compare output tensor with expected output
-        auto& net_output = layer<tag1>(net).get_output();
-        DLIB_TEST(max(abs(mat(net_output) - mat(expected_output))) < 1e-5);
-    }
-
```

Comment:
This will disappear from master when merging.

Reply:
Damn, the problem is that I can't manage to merge the master branch in my fork branches, which contains tril_... if only the tril_ tests are missing from dnn.cpp, I can put them back once all the synchronisation has been done.

Reply:
As I can't get the master version at Dlib level (i.e. with the recently integrated tril_ class), and after resolving the conflicts, I'm now trying to add the definitions for tril_ once again. If the tests pass, we'll have to merge to see if everything works correctly. If it's confirmed, I'll do the same for all the other PRs.

Reply:
Well, that seems to work. So I'm going to do the same with the other PRs

Reply:
Yeah I merge PRs by squashing the PR into one commit and adding it to master, so master will always be fine regardless.   You can (and probably should here since things have gotten complicated) squash your branch into one commit too.  Having merge commits when you are working with multiple branches is a huge pain, as you are experiencing :D  

There are a bunch of ways to squash a branch into one commit.  I like to use `git reset --soft` to just undo all the commits and then recommit them.  But there is `git merge --squash` and probably several other ways.  Make a backup branch before you do anything though if you aren't real familiar with what I'm talking about though.

Reply:
OK, but normally you can proceed because I've manually caught up the ‘mixed’ code from the different branches. When everything has been placed under the master branch, I'll remove the branches from my forks to resynchronise the fork and start again from a clean base. All that's left is to rework the last PR for softmaxm, which I'll do today.

Reply:
@davis, all the "alignments" have been made to enable the merge to go ahead.